src/Razor/src/Microsoft.AspNetCore.Razor.VSCode/src/extension.ts

```diff
@@ -182,6 +185,33 @@ export async function activate(vscodeType: typeof vscodeapi, context: ExtensionC
             localRegistrations.length = 0;
         });
 
+        const provider = new BlazorDebugConfigurationProvider(logger, vscodeType);
+        context.subscriptions.push(vscodeType.debug.registerDebugConfigurationProvider('blazor', provider));
+
+        /**
+         * On non-Windows platforms, we need to terminate the Blazor
+         * dev server and its child processes.
+         */
+        vscodeType.debug.onDidTerminateDebugSession(event => {
```

Comment:
Here's how the JS debugger does it: https://github.com/microsoft/vscode-js-debug/blob/master/src/targets/node/killTree.ts#L11 is there anything we're missing here?

Reply:
Is there a way to handle termination in our existing provider? Asking because it feels odd to have all of this logic sitting here.

Reply:
> Is there a way to handle termination in our existing provider? Asking because it feels odd to have all of this logic sitting here.

I assume that by "existing provider" you mean the new `BlazorDebugConfigurationProvider`. We could presumably register the event handler inside the `resolveDebugConfiguration` method in that provider but I'm not sure how sensible that is.

Another option is moving this event handler to its own file or exporting it from the source file for the 

> Here's how the JS debugger does it: https://github.com/microsoft/vscode-js-debug/blob/master/src/targets/node/killTree.ts#L11 is there anything we're missing here?

The biggest difference is that the JS debugger kills all children processes (including indirect descendants). I think we can get by killing just the direct child processes. That should take care of cleaning up the debug proxy and anything else the user might launch in their code.

Reply:
> Another option is moving this event handler to its own file or exporting it from the source file for the

I think i'd prefer that to keep this extension.ts as clean as possible given it's already relatively large

Reply:
> I think we can get by killing just the direct child processes.

Alright sounds good 😄

Reply:
Update: I moved the event handler to a new `Events` handle. This sets us up to add handling for more event types for the Blazor debugger in the future.

Reply:
This cleans up child processes as well.